### PR TITLE
[Jovan Mwesigwa] - Sage - Optimizing Gas Consumption in Celo Smart Contracts: A Step-by-Step Guide

### PR DESCRIPTION
**Pull request by JovanMwesigwa**
### Thanks for submitting this tutorial for Celo Sage

Before you hit "Create Pull Request", make sure you have read the [Tutorial Contribution Guidelines](https://docs.celo.org/community/celo-sage/contribution-guidelines) and reviewed the [Sample Tutorial Structure](https://docs.celo.org/community/celo-sage/tutorial-structure).

Please confirm the following before submitting your tutorial for review -

- [x] My tutorial follows the [Tutorial Contribution Guidelines](https://docs.celo.org/community/celo-sage/contribution-guidelines).
- [x] The title of the Pull Request matches the title of my tutorial.
- [x] I've followed the tutorial template structure.
- [x] I've proofread the tutorial for spelling and grammar mistakes.
- [x] I've thoroughly explained the code, with inline code comments and text explanations.
- [x] I've run a linter & prettifier to format my code blocks.
- [x] I've run the tutorial to test it and the code works.

> NOTE: We cannot review your Pull Request if it does not meet the standard outlined above.
> Reviewers - @joenyzio & @viral-sangani

### CODE REVIEW

File: blog/02-03-2023-optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide/optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide.md [hunk 1]
```diff
@@ -0,0 +1,472 @@
...
+    title: Software Engineer
+    url: https://github.com/JovanMwesigwa
+tags: [celo, 'celosage', 'intermediate', 'react', 'solidity', 'hardhat']
+hide_table_of_contents: true
+slug: /tutorials/optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide
+---
+
+![header](../../src/data-tutorials/showcase/intermediate/Optimizing_Gas_Consumption_in_Celo_Smart_Contracts.png)
+
+# Introduction​
```
Comment (Ishanpathak1) [inline blog/02-03-2023-optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide/optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide.md:17]: Change # to ##

File: blog/02-03-2023-optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide/optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide.md [hunk 2]
```diff
@@ -0,0 +1,472 @@
...
+
+Gas is the unit of currency used to pay transaction fees on the Celo blockchain. This makes optimizing gas consumption an important consideration when developing smart contracts professionally. In this tutorial, we'll explore some techniques for minimizing gas usage in Solidity smart contracts and demonstrate how to use the Hardhat gas reporter to test the efficiency of our code.
+
+# Prerequisites​
```
Comment (Ishanpathak1) [inline blog/02-03-2023-optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide/optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide.md:21]: Change # to ##

File: blog/02-03-2023-optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide/optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide.md [hunk 3]
```diff
@@ -0,0 +1,472 @@
...
+
+To follow this tutorial, you will need:
+
+* Basic knowledge of Solidity, Ethereum, and smart contracts.
+* A local development environment set up with [Node.js](https://nodejs.org/en/) and [npm](https://www.npmjs.com/), as well as the [Hardhat development environment](https://hardhat.org/getting-started/) installed. You can follow the Hardhat installation instructions to get set up.
+
+* An understanding of gas fees and how they are calculated on the Celo network. This knowledge will help us understand the optimizations we will make in this tutorial.
+
+# Requirements
```
Comment (Ishanpathak1) [inline blog/02-03-2023-optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide/optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide.md:30]: Change # to ##

File: blog/02-03-2023-optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide/optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide.md [hunk 4]
```diff
@@ -0,0 +1,472 @@
...
+
+```
+
+* An Integrated Development Environment (IDE) such as Visual Studio Code (VS Code) or any other text editor of your choice.
+
+* Basic knowledge of the command line interface (CLI) and the ability to use it to run commands.
+
+* Basic knowledge of JavaScript and familiarity with testing frameworks like Mocha and Chai.
+
+# Setting up your environment
```
Comment (Ishanpathak1) [inline blog/02-03-2023-optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide/optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide.md:51]: Change # to ##

File: blog/02-03-2023-optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide/optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide.md [hunk 5]
```diff
@@ -0,0 +1,472 @@
...
+
+Compared to the completely unoptimized contract, the report shows that the `join` function in the `MyContract` consumes `88,248` gas for one call.
+
+There is a significant reduction in the gas consumption of the join function. The previous report showed that the function consumed `325,301` of gas per call, while this report shows consumption of only `88,248` gas per call.
+
+Here's the unoptimized report:
+
+![unoptimized report](./images/first-gas-log.PNG)
+
+# Conclusion
```
Comment (Ishanpathak1) [inline blog/02-03-2023-optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide/optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide.md:452]: Change # to ##

File: blog/02-03-2023-optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide/optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide.md [hunk 6]
```diff
@@ -0,0 +1,472 @@
...
+
+In conclusion, optimizing gas usage in Solidity smart contracts is essential for reducing transaction costs and improving contract efficiency. By using techniques such as using private instead of public for constants, using view and pure functions, declaring variables with specific data types, and avoiding expensive operations such as loops, developers can significantly reduce gas usage and optimize the contract's performance. With the ever-increasing demand for blockchain-based applications, optimizing gas usage is becoming more critical than ever before, and developers should take it seriously when designing and deploying smart contracts.
+
+# Next Steps
```
Comment (Ishanpathak1) [inline blog/02-03-2023-optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide/optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide.md:456]: Change # to ##

File: blog/02-03-2023-optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide/optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide.md [hunk 7]
```diff
@@ -0,0 +1,472 @@
...
+
+After optimizing gas consumption in your Celo smart contracts with the techniques outlined in this step-by-step guide, there are several next steps you can take to continue improving the efficiency and performance of your contracts. Firstly, you can explore using the Solidity optimizer to further optimize the bytecode of your contracts. Additionally, you can consider implementing more advanced gas optimization techniques, such as using storage arrays instead of memory arrays, using structs to group related data, and leveraging the benefits of function modifiers. You can also explore using gas reporting tools and conducting regular gas audits to monitor the gas usage of your contracts over time. Finally, it's always a good idea to stay up-to-date with the latest developments in the Celo ecosystem and Solidity language to ensure you're taking advantage of the most efficient and effective techniques for optimizing gas consumption in your contracts.
+
+# About the Author​
```
Comment (Ishanpathak1) [inline blog/02-03-2023-optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide/optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide.md:460]: Change # to ##

File: blog/02-03-2023-optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide/optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide.md [hunk 8]
```diff
@@ -0,0 +1,472 @@
...
+
+Created by Jovan Mwesigwa B, Blockchain, Solidity developer
+
+Reach out:
+Twitter: [@unreal_joova](http://twitter.com/unreal_joova)
+Github: [JovanMwesigwa](https://github.com/JovanMwesigwa)
+Linkedin: [JovanMwesigwa](https://www.linkedin.com/in/jovan-mwesigwa/)
+
+# References​
```
Comment (Ishanpathak1) [inline blog/02-03-2023-optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide/optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide.md:469]: Change # to ##

File: src/data-tutorials/users.tsx [hunk 9]
```diff
@@ -1031,6 +1031,13 @@ const Users: User[] = [
     website: 'blog/tutorials/how-uniswap-works',
     tags: ['celosage', 'celo', 'beginner'],
   },
+  {
+    title: 'Optimizing Gas Consumption in Celo Smart Contracts: A Step-by-Step Guide',
+    description: 'In this comprehensive tutorial, you will learn how to optimize your smart contracts on the Celo blockchain to consume less gas. You will gain a deep understanding of gas, the key metric for measuring the cost of executing a smart contract on the Celo network. Through hands-on examples and best practices, you will discover how to write more efficient code and minimize gas usage in your Celo dApps. Whether you re a seasoned blockchain developer or just starting out, this tutorial will equip you with the skills and knowledge to build scalable, cost-effective smart contracts on the Celo platform.',
```
Comment (Ishanpathak1) [inline src/data-tutorials/users.tsx:1036]: Make description 18-20 words

File: blog/02-03-2023-optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide/optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide.md [hunk 10]
```diff
@@ -0,0 +1,472 @@
+---
+title: Optimizing Gas Consumption in Celo Smart Contracts: A Step-by-Step Guide
+description: In this comprehensive tutorial, you will learn how to optimize your smart contracts on the Celo blockchain to consume less gas. You will gain a deep understanding of gas, the key metric for measuring the cost of executing a smart contract on the Celo network. Through hands-on examples and best practices, you will discover how to write more efficient code and minimize gas usage in your Celo dApps. Whether you're a seasoned blockchain developer or just starting out, this tutorial will equip you with the skills and knowledge to build scalable, cost-effective smart contracts on the Celo platform.
```
Comment (Ishanpathak1) [inline blog/02-03-2023-optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide/optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide.md:3]: Make description 18-20 words

**Review by Ishanpathak1 — COMMENTED** [review]
Please make the required changes

File: blog/02-03-2023-optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide/optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide.md (hunk 5 above)
Comment (JovanMwesigwa) [reply blog/02-03-2023-optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide/optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide.md]: Done.

File: blog/02-03-2023-optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide/optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide.md (hunk 7 above)
Comment (JovanMwesigwa) [reply blog/02-03-2023-optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide/optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide.md]: Done

File: src/data-tutorials/users.tsx (hunk 9 above)
Comment (JovanMwesigwa) [reply src/data-tutorials/users.tsx]: Done

File: blog/02-03-2023-optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide/optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide.md (hunk 10 above)
Comment (JovanMwesigwa) [reply blog/02-03-2023-optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide/optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide.md]: Done

File: blog/02-03-2023-optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide/optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide.md (hunk 2 above)
Comment (JovanMwesigwa) [reply blog/02-03-2023-optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide/optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide.md]: Done

File: src/data-tutorials/users.tsx [hunk 11]
```diff
@@ -1031,6 +1031,13 @@ const Users: User[] = [
     website: 'blog/tutorials/how-uniswap-works',
     tags: ['celosage', 'celo', 'beginner'],
   },
+  {
+    title: 'Optimizing Gas Consumption in Celo Smart Contracts: A Step-by-Step Guide',
```
Comment (Ishanpathak1) [inline src/data-tutorials/users.tsx:1035]: Hey can you remove ":" from title, it is creating issues while deploying

File: blog/02-03-2023-optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide/optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide.md [hunk 12]
```diff
@@ -0,0 +1,472 @@
+---
+title: Optimizing Gas Consumption in Celo Smart Contracts: A Step-by-Step Guide
```
Comment (Ishanpathak1) [inline blog/02-03-2023-optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide/optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide.md:2]: Hey can you remove ":" from the title it is causing the issue while deploying

**Review by Ishanpathak1 — COMMENTED** [review]
Good work their, make the required changes and we will move this in the publish section

File: src/data-tutorials/users.tsx (hunk 11 above)
Comment (JovanMwesigwa) [reply src/data-tutorials/users.tsx]: Done

File: blog/02-03-2023-optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide/optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide.md (hunk 12 above)
Comment (JovanMwesigwa) [reply blog/02-03-2023-optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide/optimizing-gas-consumption-in-celo-smart-contracts-a-step-by-step-guide.md]: Done.

**Review by developerkunal — APPROVED** [review]
Approved @Ishanpathak1

**Pull request merged**